lifelines/utils/ShiftedBetaGeometric.py

```diff
@@ -0,0 +1,726 @@
+        opt_params = np.zeros((2 * n_params))
+
+        # clock
+        start = datetime.now()
+
+        # Print a nice looking  header for the optimization process
+        if self.verbose:
+            print('Starting Optimization with parameters:')
+            print('{:>15}: {}'.format('Samples', self.n_samples))
+            print('{:>15}: {}'.format('gamma (alpha)', self.gammaa))
+            print('{:>15}: {}'.format('gamma (beta)', self.gammaa))
+            print('{:>15}: {}'.format('Seeds', restarts))
+
+            print()
+            print("{0:^{3}} | {1:^10} | {2:13} |".format('Step',
+                                                         'Time',
+                                                         'LogLikelihood',
+                                                         print_space))
+            print("-"*35)
+
+        # Run likelihood optimization "restarts" times.
+        # noinspection PyTypeChecker
+        for step, guess in enumerate(initial_guesses):
+            # --- Variables
+            #   step: Integer - current step number
+            #  guess: Array - array with starting points for minimization
+
+            # --- Optimization
+            # Unbounded optimization (minimization) of negative log-likelihood
+            # Use a lambda function to make explicit the choice of variables.
+            # Notice that the variable p is an array of length 2 * n_params,
+            # which is split in half when passed to the _logp method. First
+            # half is alpha, second is beta.
+            new_opt = minimize(lambda p: self._logp(X=X,
+                                                    age=age,
+                                                    alive=alive,
+                                                    wa=p[:n_params],
+                                                    wb=p[n_params:]),
+                               guess,
+                               bounds=[(None, None)] * n_params * 2
+                               )
+
+            # For the first run only optimal is None, that being the case, we
+            # set the current values to both optimal (function value) as well
+            # as opt_params - parameters that minimize the function.
+            if optimal is None:
+                optimal = new_opt.fun
+                opt_params = new_opt.x
+
+            # Have we found a better value yet? If we have, update optimal
+            # with new function minimum and opt_params with corresponding
+            # minimizing parameters.
+            if new_opt.fun < optimal:
+                optimal = new_opt.fun
+                opt_params = new_opt.x
+
+            # Print current status if verbose is True.
+            if self.verbose:
+                print_string = "{step: {space}} | {elap_time:^10} | {func:13.5f} |"
+                print(print_string.format(step=step + 1,
+                                          elap_time=str(datetime.now() - start)[:-7],
```

Comment:
I prefer to use the `time.time()` instead of datetime (lighter + no need to `str`)